packages/babel-types/src/index.js

```diff
@@ -431,7 +424,6 @@ export function inherits(child: Object, parent: Object): Object {
   }
 
   t.inheritsComments(child, parent);
-  traverse.copyCache(parent, child);
```

Comment:
Agreed, the only issue would be if there's something in the cache that's critical for the functioning of some transform. I don't think there _should_ be, but that is hard to verify properly.